### PR TITLE
Sampling check

### PR DESCRIPTION
**Pull request by Amoghrd**
## Description

Check sampling > 0 instead of 1 such that the SAMPLING env var is set for eBPF pods and therefore seen on eBPF metric dashboard. It shows sampling 0 even if its set to 1 which might be misleading

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: controllers/ebpf/agent_controller.go [hunk 1]
```diff
@@ -450,7 +450,7 @@ func (c *AgentController) setEnvConfig(coll *flowslatest.FlowCollector) []corev1
 	}
 
 	sampling := coll.Spec.Agent.EBPF.Sampling
-	if sampling != nil && *sampling > 1 {
+	if sampling != nil && *sampling > 0 {
```
Comment (jpinsonneau) [inline controllers/ebpf/agent_controller.go:453]: @msherif1234 was that on purepose ?

It seems legit to me to pass sampling even when `1`

**Review by jpinsonneau — APPROVED** [review]

File: controllers/ebpf/agent_controller.go (hunk 1 above)
Comment (msherif1234) [reply controllers/ebpf/agent_controller.go:453]: that was b4 my time if u check the doc on the agent it says both 0 and 1 treated the same I pointed @Amoghrd to this when he told me about the sampling metrics

Comment (Amoghrd) [reply controllers/ebpf/agent_controller.go:453]: Is it good to merge this PR then?

Comment (jotak) [reply controllers/ebpf/agent_controller.go:453]: not very important but if we allow sampling=1 in agent then there's a little optimization to do: https://github.com/netobserv/netobserv-ebpf-agent/pull/313

**Pull request merged**